src/osx/carbon/font.cpp

```diff
@@ -324,10 +325,6 @@ void wxFontRefData::AllocIfNeeded() const
 void wxFontRefData::Alloc()
 {
     wxCHECK_RET(m_info.GetPointSize() > 0, wxT("Point size should not be zero."));
-
-    // make sure the font descriptor has been processed properly
-    // otherwise the Post Script Name may not be valid yet
-    m_info.RealizeResource();
     
     // use font caching, we cache a font with a certain size and a font with just any size for faster creation
     wxString lookupnameNoSize = wxString::Format("%s_%d_%d", m_info.GetPostScriptName(), (int)m_info.GetStyle(), m_info.GetNumericWeight());
```

Comment:
I could well be missing something, but I don't see how can this behave correctly if the PostScript name is now (almost always) empty here: these strings are cache names, so they should be unique, but now they seem to be the same for all fonts with the same style and weight. I think we should be using family/face name if PostScropt name is empty.